hazelcast/test/src/compact_test.cpp

```diff
@@ -156,7 +355,7 @@ class CompactSerializationTest : public ::testing::Test
     }
 };
 
-TEST_F(CompactSerializationTest, testAllFields)
+TEST_F(CompactSerializationTest, testAllTypes)
```

Comment:
I think that this test using `to_data_and_back_to_object` does not fully cover any bugs that can exist at `to_data` but if we recover it in `to_object` method, the test will PASS but the implementation may not be correct. We also ned to verify each part separately as well to complete the test coverage.